plugins/flatpak/gs-flatpak.c

```diff
@@ -3678,6 +3678,75 @@ app_has_local_source (GsApp *app)
 		(url != NULL && g_str_has_prefix (url, "file://"));
 }
 
+static guint64
+get_installation_dir_free_space (GsFlatpak *self)
+{
+	g_autoptr (GFile) installation_dir = NULL;
+	g_autoptr (GFileInfo) info = NULL;
+	g_autoptr (GError) local_error = NULL;
+	guint64 free_space = 0;
+
+	installation_dir = flatpak_installation_get_path (self->installation);
+
+	info = g_file_query_filesystem_info (installation_dir,
+					     G_FILE_ATTRIBUTE_FILESYSTEM_FREE,
+					     NULL,
+					     &local_error);
+	if (info == NULL)
+		g_warning ("Failed to query filesystem info: %s", local_error->message);
```

Comment:
You are not returning here, so it will proceed with a NULL `info` and (likely) crash.